### DISCUSSION TITLE
Decouple Python version build scripts from application

### DISCUSSION THREAD

**Discussion by jerr0328** [pyenv/pyenv#2327] [Ideas]
I've installed pyenv with homebrew, since it saves me a manual step when setting up a new system, and it generally works well for me (I don't have to think about updating pyenv, it happens when I periodically run `brew upgrade`). However, I noticed that I don't have the newest Python versions for a while, because I only get those when a new release is published, which can take quite some time. I understand maintainers don't want to be publishing releases every time there's a new CPython/PyPy/etc. version release, so I propose perhaps decoupling the main application from all the build scripts, similar to how homebrew is split between the core application and all the formulae. The main application could then be installed and updated with homebrew, while a separate repository could contain all the build scripts and versions, which can be more easily updated via Git (for example, with a `pyenv install --update` command that does a `git pull` on a pyenv-build-scripts repo).

This adds a bit of complexity in that now you have the risk that an app version breaks compatibility with the build scripts repo, but I'm not sure how often that would need to happen (and if it couldn't happen in some non-breaking way, possibly with some version compatibility checks on the build scripts/metadata)

**Comment by native-api** [comment]
You can install the head version of Pyenv with Homebrew with `brew install --head`.
